### PR TITLE
ON-35037 # Added `TaskGroup` and updated `ScheduledTasks` menu item

### PR DESCRIPTION
**Pull request by jdawg093**

### CODE REVIEW

File: typescript/formsApps.d.ts [hunk 1]
```diff
@@ -45,6 +45,10 @@ export type FormsAppScheduledTasksMenuItem = FormsAppBaseMenuItem & {
   tasks: Array<{ taskId: number }>
   /** If true, menu item will be the default item shown */
   isDefault: boolean
+  /** The ids of the task groups assigned to the app */
+  taskGroups: Array<{ taskGroupIds: number }>
```
Comment (divporter) [inline typescript/formsApps.d.ts:49]: I'm not sure if this enough, because it's not the group, but an instance of it.

Comment (jdawg093) [reply typescript/formsApps.d.ts:49]: ahhh you're right, it'll need a label at least too

Comment (divporter) [inline typescript/formsApps.d.ts:49]:
```suggestion
  taskGroups: Array<{ taskGroupId: number, taskGroupInstanceId: string, label: string }>
```
eg an instance of Pumper (taskGroupId 1), instanceId: abc123 (guid), and label - 'Pumper 123'

Comment (divporter) [reply typescript/formsApps.d.ts:49]: I think the API will probably need to assign the GUID so we can track task completion properly but it needs more thought (from me anyway)

Comment (divporter) [reply typescript/formsApps.d.ts:49]: I wouldn't be against a separate screen to define the group instance or item or whatever we call it.  Then we can a menu item called "Reusable Tasks" or "Repeated Tasks" whatever makes sense to the user and then have menu item called "Items" and you assign items to the app, not task groups.  To me that's the cleanest way of doing this from an E2E perspective.

File: typescript/scheduledTasks.ts [hunk 2]
```diff
@@ -89,3 +91,19 @@ export interface CompletedTask {
    */
   completedAt?: string
 }
+
+export type NewTaskGroup = {
+  /** The label of the task group */
+  label: string
```
Comment (divporter) [inline typescript/scheduledTasks.ts:97]:
```suggestion
  name: string
```

File: typescript/formsApps.d.ts [hunk 3]
```diff
@@ -47,12 +47,28 @@ export type FormsAppScheduledTasksMenuItem = FormsAppBaseMenuItem & {
   isDefault: boolean
 }
 
+export type FormsAppScheduledTasksGroupMenuItem = FormsAppBaseMenuItem & {
+  /** Type of menu item */
+  type: 'SCHEDULED_TASK_GROUPS'
+  /** The ids of the task groups assigned to the app */
+  taskGroups: Array<TaskGroupInstance>
```
Comment (divporter) [inline typescript/formsApps.d.ts:54]: Let's make this

```suggestion
  taskGroupsInstances: Array<TaskGroupInstance>
```

**Pull request merged**